parcels/gridset.py

```diff
@@ -37,10 +37,10 @@ def add_grid(self, field):
 
         if not existing_grid:
```

Comment:
Is it not the idea of the `hash` that it is much easier also to compare the grid? Would `existing_grid` not be a matter of comparing hashes, now? If so, we also don't need the if statements here, right?

Reply:
All the things we do are wrong.

We should compare the object themselves, neither a string nor a hash (it's a private variable, used to compare the objects)

I did it originally, because I was worrying of breaking something while passing the objects to C. So this was the way to compare them. This is not necessary anymore. Clean up is coming